test/test_autograd.py

```diff
@@ -5369,9 +5369,10 @@ def gradgradcheck_method_precision_override(test_name):
     return override
 
 def run_grad_and_gradgrad_checks(test_case, name, test_name, apply_method, output_variable,
-                                 input_variables, run_gradgradcheck=True, check_batched_grad=True):
+                                 input_variables, run_gradgradcheck=True, check_batched_grad=True,
```

Comment:
These changes to `test_autograd` are for method tests? Maybe you could add a similar check that checks_forward_ad=True would raise an error unless its explicitly set to True?

Reply:
This will induce larger changes because some other tests will need to be updated as well. I'll do that in a follow up PR.